pkg/command/config.go

```diff
@@ -67,21 +69,56 @@ var getConfigCmd = &cobra.Command{
 	Use:               "get",
 	Short:             "Get the current configuration",
 	ValidArgsFunction: noMoreCompletions,
+	Args:              cobra.NoArgs,
 	RunE: func(cmd *cobra.Command, args []string) error {
 		cfg, err := configlib.GetClientConfig()
 		if err != nil {
 			return err
 		}
 
+		// Print the entire config
 		b, err := yaml.Marshal(&cfg)
 		if err != nil {
 			return err
 		}
-		fmt.Println(string(b))
+		fmt.Fprintln(cmd.OutOrStdout(), strings.TrimSpace(string(b)))
+
+		warningForShadowedEnvVars(cmd.ErrOrStderr())
+
 		return nil
 	},
 }
 
+// Check if any of the variables of the config file are shadowed by
+// a variable defined in the shell.  If so, warn the user.
+func warningForShadowedEnvVars(writer io.Writer) {
+	log.SetStderr(writer)
+
+	varsInConfig := configlib.GetEnvConfigurations()
+	varNames := make([]string, 0, len(varsInConfig))
+	for k := range varsInConfig {
+		varNames = append(varNames, k)
+	}
+	sort.Strings(varNames)
+
+	first := true
+	for _, name := range varNames {
+		configValue := varsInConfig[name]
+		envValue := os.Getenv(name)
+		if envValue != configValue {
+			if first {
+				first = false
+				fmt.Fprintln(writer, "")
+				log.Warning("The following variables set in the current shell take precedence over the ones of the same name set in the tanzu config:\n")
```

Comment:
q: was there a specific reason why we this is not done with "fmt.Fprintln(writer..."?

Reply:
I just wanted to use an official warning that will be aligned with all other warnings.

But now that you mention it, this situation is not something unexpected (where we usually use a `log.Warning` message), but is a valid situation where we want to warn the user.  In such cases, we now use `Note:` as a prefix.  For example when a context is active but one or more of its plugins is not installed we show:
```
Note: As shown above, some recommended plugins have not been installed or are outdated. To install them please run 'tanzu plugin sync'.
```

I will change this PR to use `Note:` as it is more appropriate.
Thanks for bringing this to my attention.

Reply:
@vuil I left the new `Note:.....` on stderr, but I remain hesitant.  The `Note:` is not part of the config, so if a user does `tanzu config get > config.txt` it feels like the note should not be there.  However, `tanzu config get | grep Note` makes me think otherwise.

Any opinion?

Reply:
I am not sure I understand the last comment. I do think the note should note not be part of the actual output of the config get command. The note text and the list of affected vars I consider to be part of the same note. I asked the original question because I want to make sure wherever the affect var list goes the note should go with it (otherwise we are missing context as to why the list was produced in the first place)

Reply:
Let me know if you’re ok with me merging?